docs/source/tensors.rst

```diff
@@ -103,9 +103,50 @@ A tensor can be created with :attr:`requires_grad=True` so that
             [ 2.0000,  2.0000]])
 
 Each tensor has an associated :class:`torch.Storage`, which holds its data.
-The tensor class provides multi-dimensional, `strided <https://en.wikipedia.org/wiki/Stride_of_an_array>`_
+The tensor class also provides multi-dimensional, `strided <https://en.wikipedia.org/wiki/Stride_of_an_array>`_
 view of a storage and defines numeric operations on it.
 
+In general there're 3 kinds of PyTorch ops:
+
+- out-place ops: output is a new tensor with brand new storage.
+- in-place ops: output is the input tensor with modified content, in other words `id(output) == id(input)`.
+- view ops: output is a new tensor which is a view of input tensor storage.
+
+Different tensor views on the same storage allows us to do the following things:
+
+- Fast and memory efficient reshaping. For example to view any tensor a single 1D buffer, you can do `t.view(-1)`.
+- Fast and memory efficient slicing. For example to access the 2nd channel of any 3D Tensor representing an image you can do `t.select(0, 1)`.
+- Fast element-wise operations of unrelated values. If you have 20 parameters that are scalars. You can create a single Tensor that hold all of them, and each parameter is just a view into this Tensor. When you need to perform an operation on all of these, you can replace the for-loop by simply doing the operation on the full Tensor and all the scalar parameters will be updated automatically.
+
+But it can give surprising results if you are expecting a brand new Tensor. User should understand and use view ops as needed.
+
+For reference, here’s a list of all view ops in PyTorch:
+
+- :meth:`~torch.Tensor.transpose`
+- :meth:`~torch.Tensor.as_strided`
+- :meth:`~torch.Tensor.diagonal`
+- :meth:`~torch.Tensor.expand`
+- :meth:`~torch.Tensor.narrow`
+- :meth:`~torch.Tensor.permute`
+- :meth:`~torch.Tensor.select`
+- :meth:`~torch.Tensor.squeeze`
+- :meth:`~torch.Tensor.t`
+- :meth:`~torch.Tensor.unfold`
+- :meth:`~torch.Tensor.unsqueeze`
+- :meth:`~torch.Tensor.indices`
+- :meth:`~torch.Tensor.values`
+- :meth:`~torch.Tensor.narrow`
+- :meth:`~torch.Tensor.view`
+- :meth:`~torch.Tensor.view_as`
+- :meth:`~torch.Tensor.unbind`
```

Comment:
This list should also include the entries from `RETURNS_VIEWS_OF_INPUT`. Namely chunk and split as per current master.